src/ppl/src/IOPlacer.cpp

```diff
@@ -819,6 +898,109 @@ std::vector<Point> IOPlacer::findLayerSlots(const int layer, const Edge edge)
   return slots;
 }
 
+std::vector<Point> IOPlacer::findLayerSlotsPolygon(const int layer,
```

Comment:
My comments here are similar to the comment in the `checkBlockedPolygon` function. The differences between the `findLayerSlotsPolygon` and `findLayerSlots` are small and don't justify having two different functions, specially when they are very long functions.

Please merge them into a single one, using the same logic suggested previously.